content/ideas/hls-more-cabal-plugin.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Haskell Language Server: Cabal Plugin Continuation
+---
+
+The [`hls-cabal-plugin`](https://github.com/haskell/haskell-language-server/tree/master/plugins/hls-cabal-plugin) is a Haskell Language Server (HLS) plugin that allows HLS to be a Language Server for `.cabal` files as well as Haskell files.
+While the plugin already provides many core features, there are more possible features that would increase the ergonomics of working with cabal files, such as:
+
+* Goto-Definition for local stanzas, such as library, executable or common stanzas
```

Comment:
We could also potentially do something like this for non-local packages by linking to Hackage? I think the `textDocument/documentLink` request might work for that.

Reply:
Yeah, I have even more ideas, but I figured, it might be fun to let the mentee come up with their own nice features and ideas.